google-codelab/google_codelab.js

```diff
@@ -148,6 +152,12 @@ class Codelab extends HTMLElement {
 
     /** @private {?Transition} */
     this.transitionOut_ = null;
+
+    /** @private {boolean} */
+    this.resumed_ = false;
+
+    /** @private {!HTML5LocalStorage} */
+    this.storage_ = new HTML5LocalStorage();
```

Comment:
I don't know how much overhead this is, maybe nothing, but would it make sense to use some kind of a global HTML5LocalStorage instance? It's all in the same "bucket" on the browser side anyway, right?

But feel free to ignore, I haven't looked into the HTML5LocalStorage implementation at all. Maybe it was designed to be used this way.

Reply:
yeah its all browser side so not really. ITs mainly just a wrapper + browser methods around window.localStorage so making it a singelton won't gain much vs the complexity.